ipatests/test_integration/test_commands.py

```diff
@@ -1646,6 +1646,89 @@ def test_sid_generation(self):
             tasks.kinit_admin(self.master)
             self.master.run_command(['ipa', 'user-del', user])
 
+    def test_sequence_processing_ipaexternalgroup(self):
+        """Test for sequence processing failures
+
+        Issues have been found for group_add sequence processing with
+        server context. This test checks that groups have correct userclass
+        when external is set to true or false with group-add.
+
+        related: https://pagure.io/freeipa/issue/9349
+        """
+        user_code_script = textwrap.dedent("""
+            from ipalib import api, errors
+            api.bootstrap_with_global_options(context='server')
+            api.finalize()
+            if api.env.in_server:
+                api.Backend.ldap2.connect()
+            else:
+                api.Backend.rpcclient.connect()
```

Comment:
I don't think this condition is required since you are explicitly using the server context.

Reply:
I tried removing/ignoring it but then report.html showed up below error:
`AttributeError: ldap2 is not connected (ldap2_139969115393936 in MainThread)`

So just to note that now I am again adding this block of code in both of the tests.
Thank you!